hail/python/hail/experimental/ld_score_regression.py

```diff
@@ -0,0 +1,559 @@
+
+    >>> ht_gwas = hl.read_table('data/ld_score_regression.sumstats.ht')
+    >>> ht_results = hl.experimental.ld_score_regression(
+    ...     weight_expr=ht_gwas['ld_score'],
+    ...     ld_score_expr=ht_gwas['ld_score'],
+    ...     chi_squared_exprs=[ht_gwas['chi_squared_50_irnt'],
+    ...                        ht_gwas['chi_squared_20160']],
+    ...     n_samples_exprs=[ht_gwas['n_50_irnt'],
+    ...                      ht_gwas['n_20160']])
+
+
+    This function returns a :class:`Table` with one row per set of summary
+    statistics passed to the ``chi_squared_exprs`` argument. The following
+    row-indexed fields are included in the table:
+
+    *  **phenotype** (:py:data:`.tstr`) -- The name of the phenotype. The
+       returned table is keyed by this field. See the notes below for
+       details on the possible values of this field.
+    *  **mean_chi_squared** (:py:data:`.tfloat64`) -- The mean chi-squared
+       test statistic for the given phenotype.
+    *  **intercept** (`Struct`) -- Contains fields:
+
+       -  **estimate** (:py:data:`.tfloat64`) -- A point estimate of the
+          intercept :math:`1 + Na` in the LD score regression model.
+       -  **standard_error**  (:py:data:`.tfloat64`) -- An estimated
+          standard error of the intercept point estimate.
+
+    *  **snp_heritability** (`Struct`) -- Contains fields:
+
+       -  **estimate** (:py:data:`.tfloat64`) -- A point estimate of the
+          SNP-heritability :math:`h_g^2`.
+       -  **standard_error** (:py:data:`.tfloat64`) -- An estimated
+          standard error of the :math:`h_g^2` point estimate.
+
+    Notes
+    -----
+    The ``exprs`` provided as arguments to :func:`.ld_score_regression`
+    can originate from either a :class:`Table` or a :class:`MatrixTable`
+    (though they all must originate from the same object).
+
+    **If the arguments originate from a table, then:**
+
+    *  The table is assumed to be keyed by fields ``locus`` of type
+       :class:`.tlocus` and ``alleles``, a :py:data:`.tarray` of
+       :py:data:`.tstr` elements.
+    *  ``weight_expr``, ``ld_score_expr``, ``chi_squared_exprs``, and
+       ``n_samples_exprs`` are assumed to be row-indexed fields of the
+       table.
+    *  The number of expressions passed to ``n_samples_exprs`` must be
+       equal to either one or the number of expressions passed to
+       ``chi_squared_exprs``. If just one expression is passed to
+       ``n_samples_exprs``, that sample size expression is assumed to
+       apply to all sets of statistics passed to ``chi_squared_exprs``.
+       Otherwise, the first expression passed to ``n_samples_exprs``
+       will correspond to the first expression passed to
+       ``chi_squared_exprs``, the second expression passed to
+       ``n_samples_exprs`` will correspond to the second expression passed
+       to ``chi_squared_exprs``, etc.
+    *  The ``phenotype`` field that keys the table returned by
```

Comment:
I like using something more specific than `y`, since this is a genetics-specific method, not a general method like linear regression or others.

I'll change the indices to ints rather than strings.